### PR TITLE
Eventslist

### PR DESCRIPTION
**Pull request by akucharska**
**Description**

Changes proposed in this pull request:

- Introduce the EventsList widget with the possibility to define `defaultType` (ALL, NORMAL, WARNING), `hideInvolvedObjects` (`true` or `false`) and `filterBy`
- Introduce jsonata function for the most common use-case of `filterBy: "$matchByResoure($item, $root.kind, $root.metadata.name)"`,  // it is necessary since empty `filterBy` is also an option (ex. cluster nodes)
- Introduce docs and examples

**Related issue(s)**

### CODE REVIEW

File: core-ui/src/components/Extensibility/helpers/jsonataWrapper.js [hunk 1]
```diff
@@ -14,6 +14,16 @@ export function jsonataWrapper(expression) {
...
 
+  exp.registerFunction(
+    'matchByResource',
+    (resource, resourceKind, resourceName) => {
+      return (
+        resource?.involvedObject?.name === resourceName &&
+        resource?.involvedObject?.kind === resourceKind
+      );
+    },
+  );
```
Comment (Wawrzyn321) [inline core-ui/src/components/Extensibility/helpers/jsonataWrapper.js:25]: For me it's too specific to name this function `matchByResource`, while it's strongly tied to `Events` - only those use `involvedObject` with `kind` and `name`.

File: core-ui/src/components/Extensibility/components/EventList.js [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+import { EMPTY_TEXT_PLACEHOLDER } from 'shared/constants';
+
+import { ResourcesList } from 'shared/components/ResourcesList/ResourcesList';
+import { prettifyKind } from 'shared/utils/helpers';
+import { useGetTranslation } from '../helpers';
+import { jsonataWrapper } from '../helpers/jsonataWrapper';
+
+import { useMicrofrontendContext } from 'shared/contexts/MicrofrontendContext';
+
+export function EventList({
```
Comment (Wawrzyn321) [inline core-ui/src/components/Extensibility/components/EventList.js:19]: Can't we just use our built-in `EventsList`?

File: examples/resources/serverless/functions.yaml [hunk 3]
```diff
@@ -133,6 +133,12 @@ data:
                   "source": "$replicas()",
                   "name": "code.replicas",
                   "disableCreate": true
+                }, {
+                  "widget": "EventsList",
```
Comment (Wawrzyn321) [inline examples/resources/serverless/functions.yaml:137]:
```suggestion
                  "widget": "EventList",
```

File: docs/extensibility/display-section.md [hunk 4]
```diff
@@ -468,6 +469,39 @@ Columns widgets render the child widgets in multiple columns.
 
 <img src="./assets/display-widgets/Columns.png" alt="Example of a columns widget" style="border: 1px solid #D2D5D9">
 
+### EventList
+
+EventList widget renders a list of Events.
```
Comment (grego952) [inline docs/extensibility/display-section.md:474]:
```suggestion
EventList widget renders a list of events.
```

File: docs/extensibility/display-section.md [hunk 5]
```diff
@@ -468,6 +469,39 @@ Columns widgets render the child widgets in multiple columns.
...
+
+#### Widget-specific parameters
+
+- **filterBy** - A JSONata function to filter events. There is a special custom function [matchEvents](jsonata.md#matcheventsitem-kind-name) used to filter events for specific resource, for example, `$matchEvents($item, $root.kind, $root.metadata.name)`
```
Comment (grego952) [inline docs/extensibility/display-section.md:478]:
```suggestion
- **filterBy** - A JSONata function to filter events. There is a special custom function [matchEvents](jsonata.md#matcheventsitem-kind-name) used to filter events for specific resource, for example, `$matchEvents($item, $root.kind, $root.metadata.name)`.
```

File: docs/extensibility/display-section.md [hunk 6]
```diff
@@ -468,6 +469,39 @@ Columns widgets render the child widgets in multiple columns.
...
+- **defaultType** - either `ALL`, `NORMAL` or `WARNING`. When set to `NORMAL` or `WARNING` events with specific type are displayed. By default all events are fetched.
+- **hideInvolvedObjects** - optional flag. If set to `true`, the **Involved Objects** column will be hidden. Defaults to false.
```
Comment (grego952) [inline docs/extensibility/display-section.md:480]:
```suggestion
- **hideInvolvedObjects** - optional flag. If set to `true`, the **Involved Objects** column is hidden. Defaults to `false`.
```

File: docs/extensibility/jsonata.md [hunk 7]
```diff
@@ -34,6 +35,26 @@ Example from [dataSources](datasources-section.md).
     filter: '$matchByLabelSelector($item, $root.spec.selector)'
 ```
 
+## matchEvents(item, kind, name)
+
+This function can be used to match Events using a resource selector.
```
Comment (grego952) [inline docs/extensibility/jsonata.md:40]:
```suggestion
You can use this function to match events using a resource selector.
```

File: docs/extensibility/jsonata.md [hunk 8]
```diff
@@ -34,6 +35,26 @@ Example from [dataSources](datasources-section.md).
...
+
+### Function parameters
+
+- **item** - Event to check.
```
Comment (grego952) [inline docs/extensibility/jsonata.md:44]:
```suggestion
- **item** - Event to be checked.
```

File: docs/extensibility/jsonata.md [hunk 9]
```diff
@@ -34,6 +35,26 @@ Example from [dataSources](datasources-section.md).
...
+- **kind** - resource kind to match against.
```
Comment (grego952) [inline docs/extensibility/jsonata.md:45]: to match against what?

File: docs/extensibility/jsonata.md [hunk 10]
```diff
@@ -34,6 +35,26 @@ Example from [dataSources](datasources-section.md).
...
+- **name** - resource name to match against.
```
Comment (grego952) [inline docs/extensibility/jsonata.md:46]: See my previous comment

File: core-ui/src/components/Extensibility/helpers/jsonataWrapper.js (hunk 1 above)
Comment (Lyczeq) [reply core-ui/src/components/Extensibility/helpers/jsonataWrapper.js]: I changed it to `matchEvents`.

File: core-ui/src/components/Extensibility/components/EventList.js (hunk 2 above)
Comment (Lyczeq) [reply core-ui/src/components/Extensibility/components/EventList.js]: Sure we can, I did that.

File: docs/extensibility/display-section.md [hunk 11]
```diff
@@ -468,6 +469,39 @@ Columns widgets render the child widgets in multiple columns.
 
 <img src="./assets/display-widgets/Columns.png" alt="Example of a columns widget" style="border: 1px solid #D2D5D9">
 
+### EventList
+
+EventList widget renders a list of Events.
+
+#### Widget-specific parameters
+
+- **filterBy** - A JSONata function to filter events. There is a special custom function [matchEvents](jsonata.md#matcheventsitem-kind-name) used to filter events for specific resource, for example, `$matchEvents($item, $root.kind, $root.metadata.name)`.
```
Comment (grego952) [inline docs/extensibility/display-section.md:478]:
```suggestion
- **filterBy** - A JSONata function you can use to filter events emitted by a specific resource. There is a special custom function [matchEvents](jsonata.md#matcheventsitem-kind-name) used to filter events for specific resource, for example, `$matchEvents($item, $root.kind, $root.metadata.name)`.
```

File: docs/extensibility/jsonata.md [hunk 12]
```diff
@@ -34,6 +35,26 @@ Example from [dataSources](datasources-section.md).
...
 
+## matchEvents(item, kind, name)
+
+You can use this function to match Events using a resource selector.
+
+### Function parameters
+
+- **item** - Event to be checked.
+- **kind** - kind of the Event emitting resource
+- **name** - name of the event emitting resource
```
Comment (grego952) [inline docs/extensibility/jsonata.md:46]:
```suggestion
- **name** - name of the Event emitting resource
```

File: docs/extensibility/jsonata.md [hunk 13]
```diff
@@ -34,6 +35,26 @@ Example from [dataSources](datasources-section.md).
...
 ```
 
+## matchEvents(item, kind, name)
+
+You can use this function to match Events using a resource selector.
+
+### Function parameters
+
+- **item** - Event to be checked.
+- **kind** - kind of the Event emitting resource
```
Comment (grego952) [inline docs/extensibility/jsonata.md:45]:
```suggestion
- **kind** - kind of the Event emitting resource.
```

File: docs/extensibility/jsonata.md [hunk 14]
```diff
@@ -34,6 +35,26 @@ Example from [dataSources](datasources-section.md).
...
+- **name** - name of the Event emitting resource
```
Comment (grego952) [inline docs/extensibility/jsonata.md:46]:
```suggestion
- **name** - name of the Event emitting resource.
```

**Review by grego952 — DISMISSED** [review]

**Review by grego952 — APPROVED** [review]

**Review by Wawrzyn321 — DISMISSED** [review]

**Review by grego952 — APPROVED** [review]

**Review by Wawrzyn321 — APPROVED** [review]

**Pull request merged**